nix/shell.sh

```diff
@@ -2,31 +2,49 @@ echo nix/shell.sh: Entering a devShell
 export SRC_ROOT=$(pwd)
 export PATH=$SRC_ROOT/build/bin:$PATH
 
-# Function to display a formatted banner with the provided text
+# Setup Python environment using uv
+function setup_python_env {
+    VENV_DIR="$SRC_ROOT/.venv"
+    PYPROJECT_PATH="$SRC_ROOT/tests/integrationv2"
+
+    # Create virtual environment if it doesn't exist
+    if [ ! -d "$VENV_DIR" ]; then
+        echo "Creating Python virtual environment using uv..."
+        uv venv "$VENV_DIR"
+
+        # Install dependencies from pyproject.toml
+        echo "Installing Python dependencies from $PYPROJECT_PATH..."
+        uv pip install -e "$PYPROJECT_PATH"
+    fi
+
+    # Activate the virtual environment
+    echo "Activating Python virtual environment..."
+    source "$VENV_DIR/bin/activate"
+
+    # Add the virtual environment's bin directory to PATH
+    export PATH="$VENV_DIR/bin:$PATH"
+
+    echo "Python virtual environment is ready."
+}
+
+# Call the function to set up the Python environment
+setup_python_env
```

Comment:
The devShell is picking this up and activating the venv, we might need to move this inside `uvinteg` (which was proposed before...)